addons/sales_team/models/crm_team_member.py

```diff
@@ -0,0 +1,188 @@
+            for membership in self if membership.active
+        )
+        for membership in self:
+            potential = existing.filtered(lambda m: m.user_id == membership.user_id and \
+                m.crm_team_id == membership.crm_team_id and m.id != membership.id
+            )
+            if not potential or len(potential) > 1:
+                duplicates += potential
+                continue
+            if active_records.get(potential.user_id.id):
+                duplicates += potential
+            else:
+                active_records[potential.user_id.id] = potential.crm_team_id.id
+
+        if duplicates:
+            raise exceptions.UserError(
+                _("You are trying to create duplicate membership(s). We found that %(duplicates)s already exist(s).",
+                  duplicates=", ".join("%s (%s)" % (m.user_id.name, m.crm_team_id.name) for m in duplicates)
+                 ))
+
+    @api.depends('crm_team_id', 'is_membership_multi', 'user_id')
+    @api.depends_context('default_crm_team_id')
+    def _compute_user_in_teams_ids(self):
+        if all(m.is_membership_multi for m in self):
+            member_user_ids = self.env['res.users']
+        elif self.ids:
+            member_user_ids = self.env['crm.team.member'].search([('id', 'not in', self.ids)]).user_id
+        else:
+            member_user_ids = self.env['crm.team.member'].search([]).user_id
+        for member in self:
+            if member_user_ids:
+                member.user_in_teams_ids = member_user_ids
+            elif member.crm_team_id:
+                member.user_in_teams_ids = member.crm_team_id.member_ids
+            elif self.env.context.get('default_crm_team_id'):
+                member.user_in_teams_ids = self.env['crm.team'].browse(self.env.context['default_crm_team_id']).member_ids
+            else:
+                member.user_in_teams_ids = self.env['res.users']
+
+    @api.depends('crm_team_id')
+    def _compute_user_company_ids(self):
+        all_companies = self.env['res.company'].search([])
+        for member in self:
+            team_company = member.crm_team_id.company_id
+            if not team_company:
+                member.user_company_ids = all_companies
+            else:
+                member.user_company_ids = member.crm_team_id.company_id
+
+    @api.depends('crm_team_id')
+    def _compute_is_membership_multi(self):
+        multi_enabled = self.env['ir.config_parameter'].sudo().get_param('sales_team.membership_multi', False)
+        self.is_membership_multi = multi_enabled
+
+    @api.depends('is_membership_multi', 'active', 'user_id', 'crm_team_id')
+    def _compute_member_warning(self):
+        """ Display a warning message to warn user they are about to archive
+        other memberships. Only valid in mono-membership mode and take into
+        account only active memberships as we may keep several archived
+        memberships. """
+        if all(m.is_membership_multi for m in self):
```

Comment:
You check ALL here.
But if you get a mix of "multi" and "mono", your else part doesn't check it and you will get warning when you're not supposed to.

Reply:
Actually you cannot have a mix as it is a global configuration (either you support multi memberships, either not, at db level). We could simply check the config parameter, it does the same.